stanford_media.post_update.php

```diff
@@ -0,0 +1,355 @@
+          $row_group['items'][$media_browser_position] = 'DrupalMediaLibrary';
+          $editor->setSettings($settings);
+          $editor->save();
+        }
+      }
+    }
+  }
+
+  $display_modes = ['full'];
+  $media_settings = \Drupal::configFactory()
+    ->getEditable('stanford_media.settings');
+  foreach ($media_settings->get('embeddable_image_styles') as $image_style) {
+    $display_modes[] = _stanford_media_post_update_8202_image_display_mode($image_style);
+  }
+  $display_modes = array_unique($display_modes);
+  $media_settings->delete();
+
+  /** @var \Drupal\filter\FilterFormatInterface $filter */
+  foreach (FilterFormat::loadMultiple() as $filter) {
+    $filters = $filter->get('filters');
+    if (isset($filters['entity_embed'])) {
+      $filters['media_embed'] = [
+        'id' => 'media_embed',
+        'provider',
+        'media',
+        'status' => TRUE,
+        'weight' => $filters['entity_embed']['weight'],
+        'settings' => [
+          'default_view_mode' => 'full',
+          'allowed_view_modes' => array_combine($display_modes, $display_modes),
+        ],
+      ];
+      // Make sure if the filter strips html tags, we need to keep the
+      // drupal-media tag and its attributes.
+      if (isset($filters['filter_html'])) {
+        $filters['filter_html']['settings']['allowed_html'] .= ' <drupal-media data-entity-type data-entity-uuid data-align data-caption data-* alt>';
+      }
+      $filter->set('filters', $filters);
+      $filter->save();
+    }
+  }
+}
+
+/**
+ * Convert drupal-entity tags in wysiwyg's to drupal-media tags.
+ */
+function stanford_media_post_update_8202(&$sandbox) {
+  if (!isset($sandbox['count'])) {
+    $sandbox['entities'] = _stanford_media_post_update_8202_entity_list();
+    $sandbox['count'] = count($sandbox['entities']);
+  }
+
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $entity_ids = array_splice($sandbox['entities'], 0, 25);
+  foreach ($entity_ids as $item) {
+    list($entity_type, $field_name, $entity_id) = explode(':', $item);
+
+    $entity = $entity_type_manager->getStorage($entity_type)->load($entity_id);
+    $field_values = $entity->get($field_name)->getValue();
+
+    foreach ($field_values as &$field_value) {
+      $field_value['value'] = _stanford_media_post_update_8202_change_tag($field_value['value']);
+    }
+
+    $entity->set($field_name, $field_values);
+    $entity->save();
+  }
+
+  $sandbox['#finished'] = empty($sandbox['entities']) ? 1 : ($sandbox['count'] - count($sandbox['entities'])) / $sandbox['count'];
+}
+
+/**
+ * Parse the drupal-entity tag, and change it into drupal-media tag.
+ *
+ * @param string $html
+ *   Original html markup
+ *
+ * @return string
+ *   New html markup with corrected tokens.
+ */
+function _stanford_media_post_update_8202_change_tag($html) {
+  preg_match_all("/<drupal-entity.*?\/drupal-entity>/s", $html, $tokens);
```

Comment:
Do we want to assume no one is using entity embed and inserted things like nodes?

Reply:
hmm. I think its safe to say its only media. but i could tweak the regex to target only media entities